feat(core): `stack.exportValue()` can be used to solve "deadly embrace" (#12778)
aws/aws-cdk: packages/@aws-cdk/core/README.md

Nested stacks also support the use of Docker image and file assets.

> ### Breaking automatic references

Reviewer: Feels like this should be part of the section that describes [cross references](https://docs.aws.amazon.com/cdk/latest/guide/resources.html#resource_stack) in the dev guide.

Maybe we can copy the contents from the dev guide to here and just add this as a subsection? And then also duplicate this to the dev guide?

The automatic references created by CDK when you use resources across stacks
are convenient, but may block your deployments if you want to remove the
resources that are referenced in this way. You will see an error like:

```
Export Producer:ExportsOutputFnGetAtt-****** cannot be deleted as it is in use by Consumer
```

Let's say there is a Bucket in the `producer`, and the `consumer` references its
`bucket.bucketName`. You now want to remove the bucket and run into the error above.

Unblocking yourself from this is a two-step process. This is how it works:

DEPLOYMENT 1: break the relationship

- Make sure `consumer` no longer references `bucket.bucketName` (maybe the consumer

>   stack now uses its own bucket, or it writes to a Dynamo table, or maybe you just

Reviewer: ```suggestion
  stack now uses its own bucket, or it writes to an AWS DynamoDB table, or maybe you just
```

>   remove the Lambda Function altogether).

Reviewer: ```suggestion
  remove the AWS Lambda Function altogether).
```

> - In the `producer` class, call `this.exportAttribute(this.bucket.bucketName)`. This

Reviewer: ```suggestion
- In the `producer` class, call `stack.exportAttribute(this.bucket.bucketName)`. This
```

Author: People usually write in the constructor, so it's definitely `this` at that point.

in the producing stack and an
[Fn::ImportValue](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-importvalue.html)
in the consuming stack to transfer that information from one stack to the
other.

### Removing automatic cross-stack references

The automatic references created by CDK when you use resources across stacks
are convenient, but may block your deployments if you want to remove the
resources that are referenced in this way. You will see an error like:

```text
Export Stack1:ExportsOutputFnGetAtt-****** cannot be deleted as it is in use by Stack1
```

Let's say there is a Bucket in the `stack1`, and the `stack2` references its
`bucket.bucketName`. You now want to remove the bucket and run into the error above.

It's not safe to remove `stack1.bucket` while `stack2` is still using it, so
unblocking yourself from this is a two-step process. This is how it works:

DEPLOYMENT 1: break the relationship

- Make sure `stack2` no longer references `bucket.bucketName` (maybe the consumer
  stack now uses its own bucket, or it writes to an AWS DynamoDB table, or maybe you just
  remove the Lambda Function altogether).

> - In the `stack1` class, call `this.exportAttribute(this.bucket.bucketName)`. This

Reviewer 2: Seems the duplicate docs (readme + tsdoc) got out-of-sync during review/changes, e.g. this should be `exportValue`
